cmd/snap/cmd_debug_recovery_key.go

```diff
@@ -0,0 +1,58 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+// +build !nosecboot
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/secboot"
+)
+
+type cmdRecoveryKey struct{}
+
+func init() {
+	cmd := addDebugCommand("show-recovery-key",
+		"(internal) show the fde recovery key",
+		"(internal) show the fde recovery key",
+		func() flags.Commander {
+			return &cmdRecoveryKey{}
+		}, nil, nil)
+	cmd.hidden = true
+}
+
+func (x *cmdRecoveryKey) Execute(args []string) error {
+	if release.OnClassic {
+		return errors.New(`command "show-recovery-key" is not available on classic systems`)
+	}
+
+	rkey, err := secboot.RecoveryKeyFromFile(filepath.Join(dirs.SnapFDEDir, "recovery.key"))
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(Stdout, "%s\n", rkey)
```

Comment:
Is it worth providing a hint about how or where to store a backup of this key - particularly stressing that it should be backed up on a separate device or other backup medium, and that the backup medium should not be stored with this device?

Reply:
Ah, I missed the "This command is intended to be used by console-conf or similar tools" in the description, so perhaps that kind of detail should live elsewhere anyway.